### DISCUSSION TITLE
Multi select filter [v7]

### DISCUSSION THREAD

**Discussion by vsSanti** [TanStack/table#2047] [General]
I'm trying to develop a column filter that can be multi-selected. To do this, I'm using the react-select component and passing the "isMulti" props to it. The thing is: it doesn't work as expected due to setFilter column function only accepts one value.

Just for clarification: I've seen it for previous versions, but none of them work for v7.

I tried the following:
```
<Select className="react-select-container w-auto" classNamePrefix="react-select" style={{ width: '0px !important' }}
    options={options}
    isMulti
    onChange={e => {
        setFilter(e ? e.map(data => data.value) : undefined)
    }}
/>
```
But it did not work.
Does it have a solution?

**Comment by vsSanti** [comment]
What I ended up with is the following:
Added another filterType (it is shown on the filtering example) to my table, and the filter is something like that:
```
export function MultiSelectFilterFn(rows, id, filterValues) {
    if (filterValues.length === 0) return rows;

    return rows.filter(r => filterValues.includes(r.values[id]));
}
```

**Reply by TimMcCauley** [reply]
This new option using `includesSome` works for me and just landed in `7.1.0`

https://github.com/tannerlinsley/react-table/pull/2329

**Reply by jelizarovas** [reply]
I'm sorry, how do you use `includesSome`, i can't find it in the docs? Thank you!

**Reply by TimMcCauley** [reply]
I am not sure if it exists in the docs yet but how it works, p.s. make sure you are using the latest version.

```
{
          Header: "test",
          accessor: test,
          canGroupBy: true,
          Filter: MultiSelectColumnFilter,
          // https://github.com/tannerlinsley/react-table/pull/2329
          filter: 'includesSome',
          aggregate: 'count',
          Aggregated: ({ value }) => `${value} (total)`
        }
```

**Reply by ljrodriguez1** [reply]
How would you write the MultiSelectColumnFilter, i am trying but i get the error FilterValue.some is not a function

**Reply by TimMcCauley** [reply]
@ljrodriguez1 this discussion may possibly help - #2253

**Reply by ljrodriguez1** [reply]
Thank you it really helped me understand more how this filters works, i have managed to get to my own solution and is working column wize. any input in how to set the filter atribute in an GlobalFilter, i want to pass it the filter atributte includeSome

**Reply by ljrodriguez1** [reply]
@TimMcCauley do you have any idea on how to set filter: includesSome on a Global Filter?

Been figthing with this for a week and have not found anything usefull
